spec/lib/services/workflow-api-service-spec.js

```diff
@@ -376,6 +376,20 @@ describe('Http.Services.Api.Workflows', function () {
         });
     });
 
+    it('should return a sanitized workflow document', function(){
+        var workflow = {
+            id       : 'testgraphid',
+            _status  : 'pending',
+            password : 'testPassword'
```

Comment:
I'd like to see more unit tests here. Does it work if password is nested one level down? What about two, three levels down?

For example, we'll see objects like this:
```
{
    definition: {
        options: {
            <task-label>: {
                password: "foo"
            }
        }
    }
}
```